utils/message_formats/message_formats.py

```diff
@@ -0,0 +1,274 @@
+    phase_offset_imag: float = None
+
+
+@dataclass
+class RxChannel:
+    """Defines an rx_channel structure within a SequenceMetadataMessage"""
+    slice_id: int = None
+    tau_spacing: int = None
+    rx_freq: float = None
+    clrfrqflag: bool = None
+    num_ranges: int = None
+    first_range: int = None
+    range_sep: float = None
+    beam_phases: np.ndarray = None
+    lags: list[Lag] = field(default_factory=list)
+
+    def remove_all_lags(self):
+        """Remove all lag entries so the list can be refilled for the next sequence"""
+        self.lags = []
+
+    def add_lag(self, lag: Lag):
+        """Add a Lag dataclass to the message."""
+        self.lags.append(lag)
+
+
+@dataclass
+class SequenceMetadataMessage:
+    """
+    Defines a message containing metadata about a sequence of data.
+    This message format is for communication from radar_control to
+    rx_signal_processing.
+    """
+    sequence_num: int = None
+    sequence_time: float = None
+    offset_to_first_rx_sample: int = None
+    rx_rate: float = None
+    output_sample_rate: float = None
+    rx_ctr_freq: float = None
+    decimation_stages: list[DecimationStageMessage] = field(default_factory=list)
+    rx_channels: list[RxChannel] = field(default_factory=list)
+
+    def add_decimation_stage(self, stage: DecimationStageMessage):
+        """Add a decimation stage to the message."""
+        self.decimation_stages.append(stage)
+
+    def remove_all_decimation_stages(self):
+        """Remove all decimation_stage entries so the list can be refilled for the next sequence"""
+        self.decimation_stages = []
+
+    def remove_all_rx_channels(self):
+        """Remove all rx_channel entries so the list can be refilled for the next sequence"""
+        self.rx_channels = []
+
+    def add_rx_channel(self, channel: RxChannel):
+        """Add an rx_channel dataclass to the message."""
+        self.rx_channels.append(channel)
+
+
+@dataclass
+class Beam:
+    """Defines a beam structure for inclusion in an RxChannelMetadata dataclass"""
+    beam_azimuth: float = None
+    beam_num: int = None
+
+
+@dataclass
+class LagTable:
+    """Defines a ltab structure for inclusion in an RxChannelMetadata dataclass"""
+    pulse_position: list[int] = field(default_factory=list)
+    lag_num: int = None
+
+
+@dataclass
+class RxChannelMetadata:
+    """Defines an RxChannelMetadata structure for inclusion in an AveperiodMetadataMessage"""
+    slice_id: int = None
+    slice_comment: str = None
+    interfacing: str = None
+    rx_only: bool = None
+    pulse_len: int = None
+    tau_spacing: int = None
+    rx_freq: float = None
+    ptab: list[int] = field(default_factory=list)
+    sequence_encodings: list = field(default_factory=list)
+    rx_main_antennas: list[int] = field(default_factory=list)
+    rx_intf_antennas: list[int] = field(default_factory=list)
+    beams: list[Beam] = field(default_factory=list)
+    first_range: int = None
```

Comment:
Actually `first_range` gets cast to an int in data_write.py, so I think experiment_prototype.py needs to get changed. I see no reason why it would have to be an integer